### PR TITLE
CAMEL-10740 - Initial work for camel-tika component.

### PR DESCRIPTION
**Pull request by bobpaulin**
Hi,

A couple of notes about the PR.  

1) Under platforms I've included the Karaf feature but I did not include the code generated for the spring boot starter since I haven't done any testing within Boot.  

2) I have prefixed the Metadata headers generated by Tika with "Tika".  I hadn't seen this done in other projects but with the shear number of them I thought it might be useful to have some sort of namespace to prevent collisions.  Like anything else in this PR I'm open to changing it if that is considered bad practice.

3) The test document and image included are from the Tika test resources.  So my expectation is that they are licensed appropriately.  

Thanks for having such a well documented contribution process.  I found it very helpful.

- Bob

### CODE REVIEW

File: components/camel-tika/src/main/java/org/apache/camel/component/tika/TikaProducer.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+            handler.getTransformer().setOutputProperty(OutputKeys.ENCODING, encoding);
+        }
+        handler.setResult(new StreamResult(output));
+        return handler;
+    }
+
+    private Writer getOutputWriter(OutputStream output, String encoding) throws UnsupportedEncodingException {
+        if (encoding != null) {
+            return new OutputStreamWriter(output, encoding);
+        } else if (System.getProperty("os.name").toLowerCase(Locale.ROOT).startsWith("mac os x")) {
```
Comment (davsclaus) [inline components/camel-tika/src/main/java/org/apache/camel/component/tika/TikaProducer.java]: Why are you doing this specially for mac ?
Also mind that system get property is slow and its maybe better to detect the os once, in doStart and then setup that default charset to use

File: components/camel-tika/src/test/java/org/apache/camel/component/tika/TikaParseTest.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        resultEndpoint.setExpectedMessageCount(1);
+
+        resultEndpoint.expectedMessagesMatches(new Predicate() {
+            @Override
+            public boolean matches(Exchange exchange) {
+                Object body = exchange.getIn().getBody(String.class);
+                Map<String, Object> headerMap = exchange.getIn().getHeaders();
+                assertThat(body, instanceOf(String.class));
+                assertThat((String) body, containsString("<body/>"));
+                assertThat(headerMap.get("TikaContent-Type"), equalTo("application/msword"));
```
Comment (davsclaus) [inline components/camel-tika/src/test/java/org/apache/camel/component/tika/TikaParseTest.java]: I think we should use Content-Type as well as that is a common header, we have it on Exchange also, eg Exchange.CONTENT_TYPE

File: components/camel-tika/src/main/java/org/apache/camel/component/tika/TikaParseOutputFormat.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.tika;
+
+public enum TikaParseOutputFormat {
+    xml, html, text, textMain;
```
Comment (davsclaus) [inline components/camel-tika/src/main/java/org/apache/camel/component/tika/TikaParseOutputFormat.java]: What is the difference between text and textMain? Maybe you can add some details to the javadoc so we get that in the genersted documentation

File: components/camel-tika/src/main/java/org/apache/camel/component/tika/TikaEndpoint.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+import org.apache.camel.Component;
+import org.apache.camel.Consumer;
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.impl.DefaultEndpoint;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+
+@UriEndpoint(scheme = "tika", title = "Tika", syntax = "tika:operation", producerOnly = true, label = "tika")
```
Comment (davsclaus) [inline components/camel-tika/src/main/java/org/apache/camel/component/tika/TikaEndpoint.java]: Use something else than tika for label, the label can be one or more to categorize the component into groups. So maybe "transformation" is a better label?

File: components/camel-tika/pom.xml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    <description>This component integrates with Apache Tika to extract content and metadata from thousands of file types.</description>
+
+    <properties>
+        <camel.osgi.export.pkg>org.apache.camel.component.tika.*</camel.osgi.export.pkg>
+        <camel.osgi.export.service>org.apache.camel.spi.ComponentResolver;component=tika</camel.osgi.export.service>
+    </properties>
+
+    <dependencies>
+
+        <dependency>
```
Comment (davsclaus) [inline components/camel-tika/pom.xml]: Can you space this file nicely.

File: components/camel-tika/src/main/java/org/apache/camel/component/tika/TikaProducer.java (hunk 1 above)
Comment (bobpaulin) [inline components/camel-tika/src/main/java/org/apache/camel/component/tika/TikaProducer.java]: This is perhaps something unique to the TikaCLI that I pulled from here https://github.com/apache/tika/commit/ac74696e322fa93bdbec982f55226f717be66d24 .  I'm thinking perhaps the better option might be just to allow the developer to explicitly set the output content encoding via the url as follows: 
tika:parse?tikaParseOutputEncoding=UTF-8

I think that might be preferable than doing something that the developer might not be aware of.

File: components/camel-tika/src/test/java/org/apache/camel/component/tika/TikaParseTest.java (hunk 2 above)
Comment (bobpaulin) [inline components/camel-tika/src/test/java/org/apache/camel/component/tika/TikaParseTest.java]: Agreed.  I'm just going to remove the Tika prefix and pass these as headers as is.

File: components/camel-tika/src/main/java/org/apache/camel/component/tika/TikaParseOutputFormat.java (hunk 3 above)
Comment (bobpaulin) [inline components/camel-tika/src/main/java/org/apache/camel/component/tika/TikaParseOutputFormat.java]: Agree.  I will update documentation to show this distinction.

File: components/camel-tika/src/main/java/org/apache/camel/component/tika/TikaEndpoint.java (hunk 4 above)
Comment (bobpaulin) [inline components/camel-tika/src/main/java/org/apache/camel/component/tika/TikaEndpoint.java]: Agreed transformation seems fitting.

File: components/camel-tika/pom.xml (hunk 5 above)
Comment (bobpaulin) [inline components/camel-tika/pom.xml]: Yes.  See you are expecting 2 space indents.  Will fix.

**Comment by bobpaulin** [comment]
@davsclaus Thank you for taking the time to provide such detailed and valuable feedback!

**Pull request closed without merging**